src/EditorFeatures/Core.Wpf/InlineRename/InlineRenameSession.cs

```diff
@@ -277,18 +277,20 @@ private void UpdateReferenceLocationsTask(Task<IInlineRenameLocationSet> allRena
             AssertIsForeground();
 
             var asyncToken = _asyncListener.BeginAsyncOperation("UpdateReferencesTask");
-            _allRenameLocationsTask = allRenameLocationsTask;
-            allRenameLocationsTask.SafeContinueWithFromAsync(
+            _allRenameLocationsTask = allRenameLocationsTask.SafeContinueWithFromAsync(
                 async t =>
                 {
                     await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true, _cancellationTokenSource.Token);
                     _cancellationTokenSource.Token.ThrowIfCancellationRequested();
 
                     RaiseSessionSpansUpdated(t.Result.Locations.ToImmutableArray());
+
+                    return t.Result;
                 },
                 _cancellationTokenSource.Token,
                 TaskContinuationOptions.OnlyOnRanToCompletion | TaskContinuationOptions.ExecuteSynchronously,
-                TaskScheduler.Default).CompletesAsyncOperation(asyncToken);
+                TaskScheduler.Default);
+            _allRenameLocationsTask.CompletesAsyncOperation(asyncToken);
 
             UpdateConflictResolutionTask();
             QueueApplyReplacements();
```

Comment:
📝 `QueueApplyReplacements` is chained on `_allRenameLocationsTask`, but depends on data produced by `RaiseSessionSpansUpdated`. If the tasks are reordered (e.g. if a `Task.Delay` is inserted above before `RaiseSessionSpansUpdated`), the data will be processed incorrectly and lead to a failure.

Reply:
i'm not understanding.  if `QueueApplyReplacements is chained on _allRenameLocationsTask`, then shouldn't it only run when _allRenameLocationsTask completes?  And wouldn't htat only complete once that lambda completes?  Which only finishes after RaiseSessionSpansUpdated runs and finishes?  Even if something caused a delay before/after RaiseSessionSpansUpdated what would change? 

Can you clarify your explanation?

Reply:
> And wouldn't htat only complete once that lambda completes?

You are confusing  `_allRenameLocationsTask` (the field) with `allRenameLocationsTask` (the local variable). 😉

Reply:
oh boy. i'll reread, but that certainly is a confusing way to name things.